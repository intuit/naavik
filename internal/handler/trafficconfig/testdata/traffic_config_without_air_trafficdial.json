{
    "apiVersion": "admiral.io/v1",
    "kind": "TrafficConfig",
    "metadata": {
      "generation": 9,
      "labels": {
        "asset": "httpbin",
        "env": "e2e"
      },
      "name": "httpbin",
      "namespace": "admiral"
    },
    "spec": {
      "edgeService": {
        "filters": [
          {
            "name": "filter-3387192"
          },
          {
            "name": "filter--671448791"
          }
        ],
        "routes": [
          {
            "filterSelector": "filter--671448791",
            "inbound": "intuit_offeringid=com.creditkarma.networth,intuit_fdp_flowname=AddConnection;/v1/{authType}!facade_request=*;/*",
            "name": "v1",
            "outbound": "/*",
            "timeout": 5000,
            "workloadEnvSelectors": [
              "e2e",
              "e2e-1",
              "qal"
            ]
          },
          {
            "filterSelector": "filter-3387192",
            "inbound": "/health/full",
            "name": "Health Check",
            "outbound": "/health/full",
            "timeout": 5000,
            "workloadEnvSelectors": [
              "e2e",
              "e2e-1",
              "qal"
            ]
          }
        ]
      },
      "quotaGroup": {
        "appQuotaGroups": [
          {
            "associatedApps": [
              "httpbin"
            ],
            "name": "App Throttling Plan",
            "quotas": [
              {
                "algorithm": "scalable",
                "behaviour": "softThrottle",
                "keyType": "appIdOrToken",
                "maxAmount": 200,
                "name": "Total",
                "rule": "/*",
                "timePeriod": "1m"
              }
            ],
            "workloadEnvSelectors": [
              "e2e",
              "e2e-1"
            ]
          }
        ],
        "totalQuotaGroups": [
          {
            "cpuLimit": 80,
            "description": "Total capacity of service",
            "memoryLimit": 80,
            "name": "Total Throttling Plan",
            "podLevelThreshold": 100,
            "quotas": [
              {
                "algorithm": "scalable",
                "behaviour": "softThrottle",
                "keyType": "any",
                "maxAmount": 200,
                "name": "Total",
                "rule": "/*",
                "timePeriod": "1m"
              }
            ],
            "regionLevelLimit": true,
            "workloadEnvSelectors": [
              "e2e"
            ]
          }
        ]
      },
      "workloadEnvs": [
        "e2e",
        "e2e-1",
        "qal"
      ]
    }
  }