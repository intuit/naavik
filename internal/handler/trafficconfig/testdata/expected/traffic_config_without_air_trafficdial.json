{
  "hosts": [
    "e2e.httpbin.mesh",
    "e2e-1.httpbin.mesh",
    "qal.httpbin.mesh"
  ],
  "http": [
    {
      "name": "v1-e2e",
      "match": [
        {
          "authority": { "exact": "e2e.httpbin.mesh" },
          "headers": { "x-intuit-route-name": { "exact": "v1" } }
        }
      ],
      "route": [{ "destination": { "host": "e2e.httpbin.mesh" } }]
    },
    {
      "name": "v1-e2e-1",
      "match": [
        {
          "authority": { "exact": "e2e-1.httpbin.mesh" },
          "headers": { "x-intuit-route-name": { "exact": "v1" } }
        }
      ],
      "route": [{ "destination": { "host": "e2e-1.httpbin.mesh" } }]
    },
    {
      "name": "v1-qal",
      "match": [
        {
          "authority": { "exact": "qal.httpbin.mesh" },
          "headers": { "x-intuit-route-name": { "exact": "v1" } }
        }
      ],
      "route": [{ "destination": { "host": "qal.httpbin.mesh" } }]
    },
    {
      "name": "Health Check-e2e",
      "match": [
        {
          "authority": { "exact": "e2e.httpbin.mesh" },
          "headers": { "x-intuit-route-name": { "exact": "Health Check" } }
        }
      ],
      "route": [{ "destination": { "host": "e2e.httpbin.mesh" } }]
    },
    {
      "name": "Health Check-e2e-1",
      "match": [
        {
          "authority": { "exact": "e2e-1.httpbin.mesh" },
          "headers": { "x-intuit-route-name": { "exact": "Health Check" } }
        }
      ],
      "route": [{ "destination": { "host": "e2e-1.httpbin.mesh" } }]
    },
    {
      "name": "Health Check-qal",
      "match": [
        {
          "authority": { "exact": "qal.httpbin.mesh" },
          "headers": { "x-intuit-route-name": { "exact": "Health Check" } }
        }
      ],
      "route": [{ "destination": { "host": "qal.httpbin.mesh" } }]
    },
    {
      "name": "defaultall-e2e",
      "match": [
        {
          "uri": { "prefix": "/" },
          "authority": { "prefix": "e2e.httpbin.mesh" }
        }
      ],
      "route": [{ "destination": { "host": "e2e.httpbin.mesh" } }]
    },
    {
      "name": "defaultall-e2e-1",
      "match": [
        {
          "uri": { "prefix": "/" },
          "authority": { "prefix": "e2e-1.httpbin.mesh" }
        }
      ],
      "route": [{ "destination": { "host": "e2e-1.httpbin.mesh" } }]
    },
    {
      "name": "defaultall-qal",
      "match": [
        {
          "uri": { "prefix": "/" },
          "authority": { "prefix": "qal.httpbin.mesh" }
        }
      ],
      "route": [{ "destination": { "host": "qal.httpbin.mesh" } }]
    }
  ]
}
